### ISSUE TITLE
[2024-12-06]: [Trim ends only]

### ISSUE THREAD

**Issue by vagkaratzas** [JLSteenwyk/ClipKIT#60]
Could you add a mode where the tool only trims the start and end of the alignment?
There is biological information in the gaps in the middle of the sequences that we would like to keep.

**Comment by JLSteenwyk** [comment]
Hi @vagkaratzas,

Thanks for writing with your proposed enhancement. 

To make sure I understand the request correctly, can you please elaborate on what you want to be trimmed from the ends? Do you specifically want, for example, 50 sites trimmed from both ends, or is there a more quantitative approach you were thinking of?

best,

Jacob

**Comment by vagkaratzas** [comment]
Ah, to clarify, I meant clip the gaps at the ends. I mainly use clipkit in my pipeline but for now I added an alternative custom module in python that does that. You can find that here: https://github.com/vagkaratzas/proteinfamilies/blob/dev/bin/clip_ends.py
Would be nice to have all the options in one tool though ;)
For example, would be nice to combine an --ends_only parameter with -m gappy --gaps

**Comment by JLSteenwyk** [comment]
Thanks for providing the additional information. So, you are proposing only consecutive sites at the ends get removed or is there a different definition of "ends"?

For example, if using the gappy mode would result in sites 0, 2, 3, 4, 5, and 6 being trimmed, the gappy mode of trimming with the --ends_only parameter would only trim site 0, correct?

**Comment by vagkaratzas** [comment]
Exactly. And if the length is 70 and there are gappy sites at 50, 60, 68, 69, 70, then 68, 69, 70 should be also removed.

**Comment by JLSteenwyk** [comment]
Hi @vagkaratzas,

This is a cool idea and should be relatively straightforward to implement. 

I am currently traveling, and it may have to be a task I tackle in early 2025. Is this something you need immediately, or can it wait?

**Comment by JLSteenwyk** [comment]
Hi @vagkaratzas,

I appreciate your patience.

I have implemented the new feature available via PyPi as [ClipKIT, v2.4.0](https://pypi.org/project/clipkit/2.4.0/).

Use the -eo / --ends_only argument to conduct end-only trimming. Additional details are available in the documentation; see here: [https://jlsteenwyk.com/ClipKIT/advanced/index.html#ends-only](https://jlsteenwyk.com/ClipKIT/advanced/index.html#ends-only).

Thank you again for using ClipKIT, and please consider checking out some of our other software!

All the best,

Jacob

**Comment by JLSteenwyk** [comment]
That's really kind of you to create an nf-core module for ClipKIT - thank you so much!
